plugins/airtable/src/SelectDataSource.tsx

```diff
@@ -10,21 +10,31 @@ interface SelectDataSourceProps {
 }
 
 export function SelectDataSource({ collection, onSelectDataSource }: SelectDataSourceProps) {
-    const [status, setStatus] = useState<"loading-bases" | "loading-tables" | "ready" | "error">("loading-bases")
+    const [status, setStatus] = useState<"loading-bases" | "loading-tables" | "ready" | "error-bases" | "error-tables">(
```

Comment:
While it’s okay to have error states, since the Plugin won’t be functional in those cases, consider throwing in error cases and letting it run into an error boundary. That way, the happy path components will remain happy flow only. However, this approach would be different if some functions remain usable in error cases.

Reply:
That makes sense 👍